source/logging/LogQueue.cpp

```diff
@@ -25,7 +25,7 @@ std::unique_ptr<LogMessage> LogQueue::getNextLog()
     unique_lock<mutex> readLock(queueLock);
     while (logQueue.empty() && !isShutdown)
     {
-        newLogNotifier.wait(readLock);
+        newLogNotifier.wait_for(readLock, chrono::seconds(1));
```

Comment:
In the most pessimistic scenario (no log messages in queue), this setting will affect how soon after a shutdown the logger process will exit (FileLogger run loop pasted below).  As a result I think the timeout should be relatively short, perhaps 100ms eg `chrono::milliseconds(100)`.

```
void FileLogger::run()
{
    unique_lock<mutex> runLock(isRunningLock);
    isRunning = true;
    runLock.unlock();

    while (!needsShutdown)
    {
        unique_ptr<LogMessage> message = logQueue->getNextLog();

        if (NULL != message)
        {
            writeLogMessage(std::move(message));
        }
        this_thread::sleep_for(std::chrono::milliseconds(DEFAULT_WAIT_TIME_MILLISECONDS));
    }
}
```